src/gamelogic/gamemodel/Environment.hpp

```diff
@@ -0,0 +1,153 @@
+/**
+ * @file Environment.hpp
+ * @author jonas
+ * @date 01.12.20
+ * Description here TODO
+ */
+
+#ifndef SECRETBUNDESTAGSERVER_ENVIRONMENT_HPP
+#define SECRETBUNDESTAGSERVER_ENVIRONMENT_HPP
+
+#include <vector>
+#include "Board.hpp"
+#include "CardRange.hpp"
+#include "Player.hpp"
+#include "GameStateRepresentation.hpp"
+
+/**
+ * @namespace The namespace for the game model.
+ */
+namespace GameModel {
+
+    /**
+     * @class The environment class represents the whole game environment containing all game objects and entities.
+     */
+    class Environment {
+
+        private:
+            Board board{};
+            const std::vector<Player> &players;
+
+        public:
+
+            /**
+             * Main constructor for the Environment class.
+             * @param players
+             */
+            explicit Environment(const std::vector<Player> &players);
+
+            /**
+             * Get all players.
+             * @return All Players as std::shared_ptr<std::vector<Player>>.
+             */
+            [[nodiscard]] auto getPlayers() const -> const std::vector<Player> &;
+
+            /**
+             * @developer Bjoern
+             * Draw N cards from the normal card pile.
+             * @param number
+             * @return
+             */
+            auto drawNCards(std::size_t number) -> CardRange;
+
+            /**
+             * Shuffles the actual Cardpile
+             */
+            void shuffleCardPile();
+
+            /**
+             *
+             *
+             * @return
+             */
+            auto incrementElectionTracker() -> std::size_t;
+
+            /**
+             *
+             */
+            auto resetElectionTracker() -> void;
```

Comment:
Trailing Return Type bitte nur wenn es einen Return Type gibt

Reply:
ist in meinem Stand enthalten